### DISCUSSION TITLE
A question about the RankFourTensor object initialization

### DISCUSSION THREAD

**Discussion by TJT-post95** [idaholab/moose#23583] [Q&A General] [answered]
Hello
In my own app’s header file, if I take the constructor with parameter to initiate the RankFourTensor object, like `RankFourTensor C(RankFourTensorTempl<Real>::initIdentitySymmetricFour);`, the compiling will not pass. But under the moose’s header file, this declaration compiling could pass. So, is this a problem with C++ include headers?
Thanks!

**Comment by GiudGiud** [comment]
Hello

Please try this
RankFourTensor C(RankFourTensor::initIdentitySymmetricFour)

I think this is an issue with which header to include

Guillaume

**Comment by hugary1995** [comment]
Please post the corresponding error message.

**Reply by TJT-post95** [reply]
I tried `RankFourTensor _test(RankFourTensorTempl<Real>::initIdentityDeviatoric)` and `RankFourTensor _test(RankFourTensor::initIdentityDeviatoric)`, and they both could not work in my own app: `error: RankFourTensorTempl<double>::initIdentityDeviatoric is not a type` and `error: RankFourTensor::initIdentityDeviatoric is not a type`. I include the header file `#include "ComputeElasticityTensorBase.h"` to build a new child class about the elastic tensor.



![1](https://user-images.githubusercontent.com/54386614/222899787-42375b67-27d0-4187-8c39-f54503eeba97.png)
![2](https://user-images.githubusercontent.com/54386614/222899795-da62a8eb-aa3c-4f50-87fd-6da0f94a22b3.png)

**Reply by hugary1995** [reply]
#include "RankFourTensor.h"

**Reply by TJT-post95** [reply]
I give it a try, but still don’t work with the same error.😂

**Reply by maxnezdyur** [reply, marked as answer]
I think its supposed to be `RankFourTensor::IdentityDeviatoric()`

**Reply by TJT-post95** [reply]
Yep, static member function could be an alternative!
